src/learn.jl

```diff
@@ -277,16 +277,20 @@ Where...
 
 """
```

Comment:
No functional changes to this file---broke apart some helper functions to make their constituent parts accessible.

...this whole file could do with a revamp, there's some duplication and also a bunch of stuff that should probably be moved to the `metrics.jl` file, but I don't want to mess with that here.